week-4/Homework/mandatory/3-water-bottle.js

```diff
@@ -22,18 +22,33 @@ let bottle = {
   volume: 0,
   fillUp: function() {
     // calling this function should pour your bottle full (volume = 100);
+    this.volume = 100
   },
   pour: function () {
     // calling this function should increase your bottle volume by 10 unit;
+    this.volume += 10
+    if (this.volume > 100) {
+      this.volume = 100      
+    }
   },
   drink: function() {
     // calling this function should decrease your bottle volume by 10 unit;
+    this.volume -= 10
+    if (this.volume < 0) {
+      this.volume = 0
+    }
   },
   isFull: function () {
     // this function should return true if your bottle is empty;
+    if (this.volume == 100) {
+      return true
+    }

```

Comment:
Well, although it was explicitly declared, it should return false if not full.
It'd also keep the function outcome more consistent. Because now it can return either `true` or `undefined` instead of `false`.
More compact solution is simply saying `return this.volume == 100`

Reply:
There is a mistake in line 42, this function is for when the bottle is full not empty.
Its still stuck in my head that I should avoid jumping into the return statement straight away. Someone planted the idea in my head that this will make it harder to maintain the code.
But I like it though

Reply:
Yes, it was a copy-paste type.
I think the idea someone wanted to plant was going to refer to something else. There is an argument that a function should have only one return statement instead of multiple. But I'm sceptical about the benefit of that approach. I don't see much harm of having multiple exit point in a function. Well, if the function is too big it might make harder to maintain as someone told you.

Reply:
So, this is more of a coding preference, it's not a case of right and wrong?

Reply:
Yes, I believe so :). This kind of topic is subjective. I'd say it can be religious :D.
Here you can see an 11 years old question and loads of answers along with examples. It is worth reading. I'm glad that you raised this topic. I never really looked it deeply.
https://stackoverflow.com/questions/36707/should-a-function-have-only-one-return-statement

I'm coping some good points from there:

> Minimize the number of returns in each routine. It's harder to understand a routine if, reading it at the bottom, you're unaware of the possibility that it returned somewhere above.

> Use a return when it enhances readability. In certain routines, once you know the answer, you want to return it to the calling routine immediately. If the routine is defined in such a way that it doesn't require any cleanup, not returning immediately means that you have to write more code.